### PR TITLE
Add account lock/unlock support to shadow module

### PR DESCRIPTION
**Pull request by czee**
### What does this PR do?

Add account lock/unlock support to shadow module
### What issues does this PR fix or reference?

None
### New Behavior

The shadow module can lock and unlock accounts
### Tests written?

No

### CODE REVIEW

File: salt/modules/shadow.py [hunk 1]
```diff
@@ -202,6 +202,54 @@ def del_password(name):
     return not uinfo['passwd']
 
 
+def lock_password(name):
+    '''
+    .. versionadded:: 2016.3.2
```
Comment (rallytime) [inline salt/modules/shadow.py]: Can you change the `2016.3.2` value to `Carbon`? That is the code name for the next feature release, which will be based off of the current develop branch.

File: salt/modules/shadow.py [hunk 2]
```diff
@@ -202,6 +202,54 @@ def del_password(name):
...
+    __salt__['cmd.run'](cmd, python_shell=False)
+
+    post_info = info(name)
+
+    return post_info['passwd'][0] == '!'
+
+
+def unlock_password(name):
+    '''
+    .. versionadded:: 2016.3.2
```
Comment (rallytime) [inline salt/modules/shadow.py]: `Carbon` here as well.

**Comment by rallytime** [comment]
@czee Thanks for adding these new functions. I had some small comments above that should be fixed, but this otherwise looks good.

**Comment by czee** [comment]
@rallytime I've updated the commit now.

**Pull request merged**